### ISSUE TITLE
High client network latencies? (local server)

### ISSUE THREAD

**Issue by AntumDeluge** [minetest/minetest#11195]
Minetest client v5.4.0
OSes:
- Windows 10 64-bit
- Android 10 (Lineage OS 17.1) 32-bit

After playing for a few minutes on my local server, my client becomes very laggy. As I haven't seen anyone else complaining about this, I'm wondering if a mod is causing it. Is there a way to debug it to find out? Is disabling each mod one by one the only way to find out?

**Comment by sfan5** [comment]
It's not entirely clear but this might be related: #11118

> Is there a way to debug it to find out?

I'd first look at the debug log to see if anything points to problems.

Please also answer these:
* Do you experience this issue on Android or on Windows? (the client matters here)
* What does "laggy" mean? Low FPS? High latency? something else?
* Which game are you playing? With which set of mods?

**Comment by AntumDeluge** [comment]
Sorry I wasn't clear. I experience high latency (if I understand latency correctly) on both Windows & Android clients. But only after I have been playing for 10-30 minutes (estimated).

I'm using my own [Antum game](https://github.com/AntumMT/game-antum) (included mods are listed in README).

**Comment by AntumDeluge** [comment]
I made a video to demonstrate the issue, but waiting for it to upload to YouTube at the moment. Should be done in 10-15 minutes.

I really don't know if my problem has anything to do with latency or lag. The problem occurs whether or not I am connected over my local network or I am using a client on the same machine that is hosting the server.

**Comment by AntumDeluge** [comment]
Here is the video: https://youtu.be/y1FSmPAbJWM

It's between 3-4 minutes long. Don't have to watch the whole thing. Performance starts to be affected around 59 seconds into it & gets progressively worse from there. At the end, I log out & back in again to show that it resets when doing so.

**Comment by SmallJoker** [comment]
Are you running out of free RAM?

**Comment by AntumDeluge** [comment]
> Are you running out of free RAM?

I haven't been able to reproduce the issue. Had my client on all night with my resources monitor up. No performance degradation.

But, I'm pretty sure I now know what the problem was. The only major change that I have made in the last 1-2 days was that I removed an old WIP mod named [mthudclock](https://forum.minetest.net/viewtopic.php?t=11879) from my server. It appears this was the offending mod. I haven't looked through the code, but I suspect there is something in there causing a memory leak.

Unless the issue resurfaces, I will close this within the next 24 hours.
